candle-pyo3/src/utils.rs

```diff
@@ -0,0 +1,40 @@
+use ::candle::{Error as CandleError, Result as CandleResult};
+use candle::Shape;
+
+/// Tries to broadcast the `rhs` shape into the `lhs` shape.
+pub fn broadcast_shapes(lhs: &::candle::Tensor, rhs: &::candle::Tensor) -> CandleResult<Shape> {
```

Comment:
Could you add a comment that explains how this differs from `broadcast_shape_binary_op`? Looks a bit like a cut and paste of the code, we could make it `pub` rather than `pub(crate)` if it's helpful here.

Reply:
Well there isn't any difference. We should probably make it `pub` as i have to reuse it here to broadcast the `cmp` methods.

Reply:
Yeah seems far better than just copying the code.